File: src/org/r3/db/bbs/bbstopic/BBSTopicDTO.java
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//


package org.r3.db.bbs.bbstopic;

public class BBSTopicDTO extends BBSTopicDTOBase
{
    public BBSTopicDTO()
    {
        super();
    }

    public BBSTopicDTO(BBSTopicDTO another)
    {
        super(another);
    }

    @Override
    public String getDtokey()
    {
        StringBuilder stb = new StringBuilder(); 
        stb.append(site).append('_');
        stb.append(topicid);
        return stb.toString();
    }
}

File: WebContent/org/r3/db/bbs/bbstopic/BBSTopicDTO.js
//
// GPA Gruppo Progetti Avanzati s.r.l. - 
// Progetto UNEP 2012 per PosteItaliane S.p.a.
//
//
// TODO: Readme Section.
//



jQuery.r3Utils.namespace("org.r3.db.bbs.bbstopic");

org.r3.db.bbs.bbstopic.BBSTopicDTO = Class.extend(
{
	 init: function(aDataBean)
	 {  
	 	 	 this.dtrowclass   = "grade_bbstopic";
       this.dtokey       = aDataBean.dtokey   || null;
       this.dtodescr     = aDataBean.dtodescr || null;
       
        this.site = aDataBean.site || null;
        this.topicid = aDataBean.topicid || null;
        this.topictitle = aDataBean.topictitle || null;
        this.cardid = aDataBean.cardid || null;
        this.moderatorid = aDataBean.moderatorid || null;
        this.moderatormode = aDataBean.moderatormode || null;
        this.options = aDataBean.options || null;
        this.recstatus = aDataBean.recstatus || null;
        this.rectimestamp = aDataBean.rectimestamp || null;
 
	 }
    
  ,compareTo: function(anItemId)
   {
   	   if (!this.dtokey)
   	       return -99;
   	       
   	   if (this.dtokey < anItemId)
   	       return -1;
   	   else if (this.dtokey > anItemId)
   	   	   return  1;
   	   	   
   	   return 0;
   }
   
});
